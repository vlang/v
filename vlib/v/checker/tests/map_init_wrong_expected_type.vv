struct Foo {
	m map[string]string
}

struct Bar {
	m map[string]int
}

fn main() {
	a := Bar{
		m: {
			'a': 5
			'b': 6
		} // ok
	}
	b := Bar{
		m: {
			'a': '5'
			'b': '6'
		} // bad
	}
	c := Bar{
		m: {
			...a.m
		} // ok
	}
	d := Bar{
		m: {
			...a.m
			'c': 7
		} // ok
	}
	e := Bar{
		m: {
			...a.m
			'c': '7'
		} // bad values
	}

	x := Foo{}
	f := Bar{
		m: {
			...x.m
		} // bad update
	}
	g := Bar{
		m: {
			...x.m
			'c': 7
		} // bad update, ok values
	}
	h := Bar{
		m: {
			...x.m
			'c': '7'
		} // bad update, bad values
	}
	println(a)
	println(b)
	println(c)
	println(d)
	println(e)
	println(f)
	println(g)
	println(h)
}
