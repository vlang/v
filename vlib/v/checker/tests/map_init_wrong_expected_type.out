vlib/v/checker/tests/map_init_wrong_expected_type.vv:18:9: error: invalid map value: expected `int`, not `string`
   16 |     b := Bar{
   17 |         m: {
   18 |             'a': '5'
      |                  ~~~
   19 |             'b': '6'
   20 |         } // bad
vlib/v/checker/tests/map_init_wrong_expected_type.vv:19:9: error: invalid map value: expected `int`, not `string`
   17 |         m: {
   18 |             'a': '5'
   19 |             'b': '6'
      |                  ~~~
   20 |         } // bad
   21 |     }
vlib/v/checker/tests/map_init_wrong_expected_type.vv:36:9: error: invalid map value: expected `int`, not `string`
   34 |         m: {
   35 |             ...a.m
   36 |             'c': '7'
      |                  ~~~
   37 |         } // bad values
   38 |     }
vlib/v/checker/tests/map_init_wrong_expected_type.vv:43:9: error: invalid map update: expected `map[string]int`, not `map[string]string`
   41 |     f := Bar{
   42 |         m: {
   43 |             ...x.m
      |                  ^
   44 |         } // bad update
   45 |     }
vlib/v/checker/tests/map_init_wrong_expected_type.vv:48:9: error: invalid map update: expected `map[string]int`, not `map[string]string`
   46 |     g := Bar{
   47 |         m: {
   48 |             ...x.m
      |                  ^
   49 |             'c': 7
   50 |         } // bad update, ok values
vlib/v/checker/tests/map_init_wrong_expected_type.vv:54:9: error: invalid map update: expected `map[string]int`, not `map[string]string`
   52 |     h := Bar{
   53 |         m: {
   54 |             ...x.m
      |                  ^
   55 |             'c': '7'
   56 |         } // bad update, bad values
vlib/v/checker/tests/map_init_wrong_expected_type.vv:55:9: error: invalid map value: expected `int`, not `string`
   53 |         m: {
   54 |             ...x.m
   55 |             'c': '7'
      |                  ~~~
   56 |         } // bad update, bad values
   57 |     }
